graphql_compiler/compiler/emit_sql.py

```diff
@@ -315,17 +424,69 @@ def _construct_fold_subquery(self, subquery_from_clause):
         ).select_from(
             subquery_from_clause
         )
-        # Factor out GROUP BY because MSSQL won't use it
-        return select_statement.group_by(
-            *self.group_by
-        )
+
+        if isinstance(self._sqlalchemy_compiler, MSDialect):
+            # mssql doesn't rely on a group by
+            return select_statement
+        elif isinstance(self._sqlalchemy_compiler, PGDialect):
+            return select_statement.group_by(
+                *self.group_by
+            )
+        else:
+            raise NotImplementedError(
+                u'Fold only supported for MSSQL and '
+                u'PostgreSQL, dialect was set to {}'.format(self._sqlalchemy_compiler.name)
+            )
 
     def _get_array_agg_column(self, intermediate_fold_output_name, fold_output_field):
         """Select an array_agg of the fold output field, labeled as requested."""
         return sqlalchemy.func.array_agg(
             self.output_vertex_alias.c[fold_output_field]
         ).label(intermediate_fold_output_name)
 
+    def _get_mssql_xml_path_column(self,
+                                   intermediate_fold_output_name,
+                                   fold_output_field,
+                                   last_traversal):
+        """Select the MSSQL XML PATH aggregation of the fold output field, labeled as requested.
+
+        The MSSQL equivalent of array aggregation is performed using an XML PATH subquery that has
+        the basic structure outlined below.
+
+        SELECT
+            COALESCE('|' + ENCODE(OutputVertex.output_field), '~')
+        FROM
+            OutputVertex
+        WHERE
+            VertexPrecedingOutput.primary_key = OutputVertex.foreign_key
+        FOR XML PATH ('')
+
+        - ENCODE is shorthand for a function composition which replaces '~' (null),
+        '|' (list delimiter), '^' (escape) with '^n', '^d', '^e'.
+
+        - VertexPrecedingOutput is the vertex immediately preceding the output vertex in the
+        chain of traversals beginning at the vertex immediately outside the fold.
+        VertexPrecedingOutput is the `from_table` of the last traversal descriptor tuple
+        added to the fold's `traversal_descriptors` list.
+
+        - The join predicate may have primary_key and foreign_key reversed depending on the
+        direction of the edge connecting VertexPrecedingOutput to OutputVertex.
+
+        Args:
+            intermediate_fold_output_name: String label to give to the resulting XML PATH
```

Comment:
nit: lowercase `String` here and elsewhere since the Python type you are referring to is also in lowercase (`str`).